src/strawberry_sqlalchemy_mapper/relay.py

```diff
@@ -0,0 +1,427 @@
+        converters = [getattr(model, attr).type.python_type for attr in attrs]
+        filters = [
+            and_(
+                *(
+                    getattr(model, attr) == converters[i](part)
+                    for i, (attr, part) in enumerate(zip(attrs, node_id.split("|")))
+                )
+            )
+            for node_id in node_ids
+        ]
+        query = query.where(or_(*filters))
+
+    return query
+
+
+@overload
+def resolve_model_node(
+    source: Union[
+        Type[WithStrawberrySQLAlchemyObjectDefinition],
+        Type[relay.Node],
+        Type[_T],
+    ],
+    node_id: Union[str, relay.GlobalID],
+    *,
+    session: Session,
+    info: Optional[Info] = ...,
+    required: Literal[False] = ...,
+    filter_perms: bool = False,
+) -> AwaitableOrValue[Optional[_T]]:
+    ...
+
+
+@overload
+def resolve_model_node(
+    source: Union[
+        Type[WithStrawberrySQLAlchemyObjectDefinition],
+        Type[relay.Node],
+        Type[_T],
+    ],
+    node_id: Union[str, relay.GlobalID],
+    *,
+    session: Session,
+    info: Optional[Info] = ...,
+    required: Literal[True],
+    filter_perms: bool = False,
+) -> AwaitableOrValue[_T]:
+    ...
+
+
+def resolve_model_node(
+    source,
+    node_id,
+    *,
+    session: Session,
+    info: Optional[Info] = None,
+    required=False,
+    filter_perms=False,
+):
+    """
+    Resolve model node.
+
+    Args:
+    ----
+        source:
+            The source model or the model type that implements the `Node` interface
+        node_id:
+            The node it to retrieve the model from
+        session:
+            A sqlalchemy session object.
+        info:
+            Optional gql execution info. Make sure to always provide this or
+            otherwise, the queryset cannot be optimized in case DjangoOptimizerExtension
+            is enabled. This will also be used for `is_awaitable` check.
+        required:
+            If the return value is required to exist. If true, `qs.get()` will be
+            used, which might raise `model.DoesNotExist` error if the node doesn't
+            exist. Otherwise, `qs.first()` will be used, which might return None.
+
+    Returns:
+    -------
+        The resolved node, already prefetched from the database
+
+    """
+    from strawberry_sqlalchemy_mapper.mapper import StrawberrySQLAlchemyType
+
+    definition = StrawberrySQLAlchemyType[Any].from_type(source, strict=True)
+    model = definition.model
+
+    if isinstance(node_id, relay.GlobalID):
+        node_id = node_id.node_id
+
+    attrs = cast(relay.Node, source).resolve_id_attr().split("|")
+    converters = [getattr(model, attr).type.python_type for attr in attrs]
+
+    filter = dict(
+        zip(
+            cast(relay.Node, source).resolve_id_attr().split("|"),
+            (converters[i](nid) for i, nid in enumerate(node_id.split("|"))),
+        )
+    )
+
+    try:
+        return session.get(model, filter)
```

Comment:
Again, we should be clear about where async is supported. Not a blocker for first version, we can add async everywhere later.